tests/unit/array/mixins/test_find.py

```diff
@@ -145,6 +145,65 @@ def test_find_by_text(storage, config, start_storage):
     assert len(results[1]) == 0  # 'token' is not present in da vocabulary
 
 
+@pytest.mark.parametrize(
+    'storage, config, filter',
+    [
+        (
+            'elasticsearch',
+            {'n_dim': 32, 'columns': {'i': 'int'}, 'index_text': True},
+            None,
+        ),
+        (
+            'elasticsearch',
+            {'n_dim': 32, 'columns': {'i': 'int'}, 'index_text': True},
+            {
+                'range': {
+                    'i': {
+                        'lte': 5,
+                    }
+                }
+            },
+        ),
+        (
+            'elasticsearch',
+            {'n_dim': 32, 'columns': {'i': 'int'}, 'index_text': True},
+            [
+                {
+                    'range': {
+                        'i': {
+                            'lte': 5,
+                        }
+                    }
+                }
+            ],
+        ),
+        ('redis', {'n_dim': 32, 'columns': {'i': 'int'}, 'index_text': True}, None),
+        (
+            'redis',
+            {'n_dim': 32, 'columns': {'i': 'int'}, 'index_text': True},
+            '@i:[-inf 5]',
+        ),
+    ],
+)
+def test_find_by_text_and_filter(storage, config, filter, start_storage):
+    da = DocumentArray(storage=storage, config=config)
+    with da:
+        da.extend(
+            [Document(id=f'{i}', tags={'i': i}, text=f'pizza {i}') for i in range(10)]
+        )
+        da.extend(
+            [
+                Document(id=f'{i+10}', tags={'i': i}, text=f'noodles {i}')
+                for i in range(10)
+            ]
+        )
+
+    results = da.find('pizza', filter=filter)
+
+    assert all([int(r.id) < 10 for r in results]) == True
```

Comment:
first u need to assrt length of results is larger than 0, otherwise I think the next asserts would still work